### ISSUE TITLE
Restoring focus outlines on form elements in Firefox

### ISSUE THREAD

**Issue by TobyBelch** [necolas/normalize.css#627]
I was using 3.0.3, and was lacking focus outlines around my form elements (in Firefox only).
I noticed this new style in 4.1.1:

`Restore the focus styles unset by the previous rule.`

`button:-moz-focusring,`
`[type="button"]:-moz-focusring,`
`[type="reset"]:-moz-focusring,`
`[type="submit"]:-moz-focusring {`
`outline: 1px dotted ButtonText;`
`}`

This fixed the problem _only_ on my Submit button, however I was still lacking focus outlines on my input, select, and textarea elements.

I extended this style as follows and now have focus outlines on _all_ my form elements:

`textarea:-moz-focusring,`
`select:-moz-focusring,`
`input:-moz-focusring,`
`button:-moz-focusring,`
`[type="button"]:-moz-focusring,`
`[type="reset"]:-moz-focusring,`
`[type="submit"]:-moz-focusring {`
`outline: 1px dotted ButtonText;`
`}`

I still don't understand why the outlines were missing in the first place.

P.S.—The value ButtonText is deprecated according to the W3C validator, but can be replaced by the equivalent:
`outline-style:  dotted;`
`outline-width: 1px;`
I assume we _don't_ want to set the outline'scolor to a fixed value, but let it inherit its color from the element's color.

**Comment by slavanga** [comment]
@TobyBelch input, select, and textarea elements don't use the same outline style as buttons so they don't need to be normalised in this case.

**Comment by hd4ng** [comment]
@slavanga Why only the firefox has this focusing outline?
